robottelo/helpers.py

```diff
@@ -366,6 +366,26 @@ def form_repo_path(org=None, lce=None, cv=None, cvv=None, prod=None, repo=None,
     return os.path.join(PULP_PUBLISHED_YUM_REPOS_PATH, repo_path)
 
 
+def form_repo_url(capsule, org=None, lce=None, cv=None, prod=None, repo=None):
+    """Forms url of a repo or CV published on a Satellite or Capsule.
+
+    :param object capsule: Capsule or Satellite object providing its url
+    :param str org: organization label
+    :param str lce: lifecycle environment label
+    :param str cv: content view label
+    :param str prod: product label
+    :param str repo: repository label
+    :return: url of the specific repo or CV
+    """
+    if not all([capsule, org, prod, repo]):
```

Comment:
This type of validation is in a way built into the language itself.

These arguments are `optional` because they have a default assignment in the function signature - `org=None`.

By omitting the default assignment (like the `capsule` kwarg), calling the function without one of its required args/kwargs will produce a `TypeError` describing the keyword or positional argument that's missing. 

You'll want to resort to checks like this only when specific combinations of arguments are required - for example `repo` is required only when `prod` is not empty.

I won't consider this a blocker as its not harmful - its just there's a more pythonic way of expressing the requirement to pass these arguments for the function to work.

Reply:
Thank you for the explanation! So in this case I could use just `def form_repo_url(capsule, org, prod, repo, lce=None, cv=None):` and remove the `if not all` condition, right?

Ok, I'll update it in the next PR.